contracts/src/PBHVerifier.sol

```diff
@@ -1,32 +1,154 @@
-    ///      to initialize the logic contract.
+    /// @notice Emitted once for each successful PBH verification.
     ///
-    /// @param _logic The initial implementation (delegate) of the contract that this acts as a proxy
-    ///        for.
-    /// @param _data If this is non-empty, it is used as the data for a `delegatecall` to `_logic`.
-    ///        This is usually an encoded function call, and allows for initialising the storage of
-    ///        the proxy in a way similar to a traditional solidity constructor.
-    constructor(address _logic, bytes memory _data) payable WorldIDProxy(_logic, _data) {
-        // !!!! DO NOT PUT PROGRAM LOGIC HERE !!!!
-        // It should go in the `initialize` function of the delegate instead.
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param payload The zero-knowledge proof that demonstrates the claimer is registered with World ID.
+    event PBH(address indexed sender, uint256 indexed nonce, bytes callData, PBHPayload payload);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Vars                                  ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev The World ID group ID (always 1)
+    uint256 internal constant _GROUP_ID = 1;
+
+    /// @dev The World ID instance that will be used for verifying proofs
+    IWorldIDGroups internal _worldId;
+
+    /// @dev The EntryPoint where Aggregated PBH Bundles will be proxied to.
+    IEntryPoint internal _entryPoint;
+
+    /// @notice The number of PBH transactions that may be used by a single
+    ///         World ID in a given month.
+    uint8 public numPbhPerMonth;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Mappings                              ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @dev Whether a nullifier hash has been used already. Used to guarantee an action is only performed once by a single person
+    mapping(uint256 => bool) public nullifierHashes;
+
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                                  Functions                             ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @param sender The sender of this particular transaction or UserOp.
+    /// @param nonce Transaction/UserOp nonce.
+    /// @param callData Transaction/UserOp call data.
+    /// @param pbhPayload The PBH payload containing the proof data.
+    function verifyPbhProof(address sender, uint256 nonce, bytes calldata callData, PBHPayload memory pbhPayload)
```

Comment:
nit: consider changing this function name to `verifyPbh` as this is not always verifying the proof. This might be confusing to anyone that only sees this function called.